protocol/src/main/java/org/opensearch/sql/protocol/response/format/ErrorFormatter.java

```diff
@@ -43,14 +43,22 @@ public static  String prettyFormat(Throwable t) {
     return prettyJsonify(error);
   }
 
+  /**
+   * Util method to format {@link Object} to string in compact printing.
+   */
   public static String compactJsonify(Object jsonObject) {
     return AccessController.doPrivileged(
-        (PrivilegedAction<String>) () -> GSON.toJson(jsonObject));
+        (PrivilegedAction<String>) () -> GSON.toJson(jsonObject)
+                .replaceAll("\\n|\\r\\n", System.getProperty("line.separator")));
```

Comment:
I think only \\n is needed here, since GSON always outputs \n to be the same across systems, so not sure if I should leave it like this

Reply:
would this modify newline in JSON string values?

Reply:
Yes It will, I was going for consistency with how the regular responses are formatted: 
https://github.com/opensearch-project/sql/blob/2.x/protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseFormatter.java#L59
which also seems to be dependent on operating system, but open to discussion about this

Reply:
i don't think it changes newline in JSON string values (user data) though, it's quoting them if they contain `,` and joining the whole thing with the platform dependent new line character
https://github.com/opensearch-project/sql/blob/9df2acd6e65f4dd0ca59a806cfe8de047f343033/protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseFormatter.java#L84

For example someone uses
```json
{
  "key": "multi \n line \n data",
  "key2": "one line data"
}
```

this can become `{\r\n  "key": "...",\r\n  "key2": "..."\r\n}`, but they might not want `multi \n line \n data` to become `multi \r\n line \r\n data`?

Reply:
is it possible to expect `\n` in the related tests?